Adds Support to `datetime.timedelta` , `astropy.units.Quantity` in `sunpy.time.TimeRange.shift` (#7931)
sunpy/sunpy: changelog/7931.feature.rst

> Now :meth:`~sunpy.time.TimeRange.extend` supports inputs of type `datetime.timedelta` and `astropy.units.Quantity`g.

Reviewer: ```suggestion
:meth:`~sunpy.time.TimeRange.extend` can now accept inputs of type `datetime.timedelta` and `astropy.units.Quantity`.
```

> :meth:`~sunpy.time.TimeRange.shift` can now accept inputs of type `datetime.timedelta` and `astropy.units.Quantity`.

Reviewer: ```suggestion
:meth:`sunpy.time.TimeRange.shift` can now accept inputs of type `datetime.timedelta` and `astropy.units.Quantity`.
```

> :meth:`~sunpy.time.TimeRange.extend` can now accept inputs of type `datetime.timedelta` and `astropy.units.Quantity`.

Reviewer 2: ```suggestion
:meth:`~sunpy.time.TimeRange.shift` can now accept inputs of type `datetime.timedelta` and `astropy.units.Quantity`.
```
